browser-test/test.endtoend.ts

```diff
@@ -0,0 +1,546 @@
+          const client = new EchoClient({
+            auth: authStub,
+            credentials: {client_email: 'bogus', private_key: 'bogus'},
+            projectId: 'bogus',
+          });
+  
+          // Mock request
+          const request = {};
+  
+          // Mock response
+          const content = 'content951530617';
+          const expectedResponse = {
+            content,
+          };
+  
+          // Mock Grpc layer
+          client._innerApiCalls.wait = mockLongRunningGrpcMethod(
+            request,
+            expectedResponse
+          );
+  
+          client
+            .wait(request)
+            .then(responses => {
+              const operation = responses[0];
+              return operation.promise();
+            })
+            .then(responses => {
+              assert.deepStrictEqual(responses[0], expectedResponse);
+              done();
+            })
+            .catch(err => {
+              done(err);
+            });
+        });
+  
+        it('invokes wait with error', done => {
+          const client = new EchoClient({
+            auth: authStub,
+            credentials: {client_email: 'bogus', private_key: 'bogus'},
+            projectId: 'bogus',
+          });
+  
+          // Mock request
+          const request = {};
+  
+          // Mock Grpc layer
+          client._innerApiCalls.wait = mockLongRunningGrpcMethod(
+            request,
+            null,
+            error
+          );
+  
+          client
+            .wait(request)
+            .then(responses => {
+              const operation = responses[0];
+              return operation.promise();
+            })
+            .then(() => {
+              assert.fail();
+            })
+            .catch(err => {
+              assert(err instanceof Error);
+              assert.strictEqual(err.code, FAKE_STATUS_CODE);
+              done();
+            });
+        });
+  
+        it('has longrunning decoder functions', () => {
+          const client = new EchoClient({
+            auth: authStub,
+            credentials: {client_email: 'bogus', private_key: 'bogus'},
+            projectId: 'bogus',
+          });
+          assert(
+            client._descriptors.longrunning.wait.responseDecoder instanceof Function
+          );
+          assert(
+            client._descriptors.longrunning.wait.metadataDecoder instanceof Function
+          );
+        });
+    });
+});
+
+describe('Run tests against gRPC server', () => {
+    const authStub = {
+        getRequestHeaders() {
+        return {Authorization: 'Bearer SOME_TOKEN'};
+        },
+      };
+    
+    const opts = {
+          auth: authStub,
+    };
+
+    const client = new EchoClient(opts);
+
+    before(async function () {
+        this.timeout(30000);
+        console.log('before block started');
+        const request = {
+            content: 'test',
+        };
+        
+        await sleep(5000);
```

Comment:
let's add // TODO here so that we don't forget about this. this is a general practice when we commit some code we plan to change later.